sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -166,6 +183,31 @@ export interface KeyVaultSelectiveKeyRestoreResult {
     startTime: Date;
 }
 
+// @public
+export interface KeyVaultSetting {
+    name: string;
+    type?: KeyVaultSettingType;
+    value: string;
```

Comment:
In other languages ([Java](https://github.com/Azure/azure-sdk-for-java/pull/31705/files#diff-0347424f525579f19cd1714a3a0da6eca5347f00059f93912c3402b21fa157e8R51), [.NET](https://github.com/Azure/azure-sdk-for-net/pull/31984/files#diff-5cf1c684ded136842e0bb7702bdfd81e9e6f24c9f0c3e3cfa69690ec2f563b50R17)), we are adding helper methods which can convert the setting's value to the type specified by the `type` property (only boolean at this stage). I'm struggling to come up with an idiomatic analogue in JS. Do we have any prior art which does a similar thing? Maybe we could leverage the type system somehow.

Reply:
Trying out a discriminated union after talking with Jose.